velox/type/tests/VariantTest.cpp

```diff
@@ -272,3 +272,22 @@ TEST(VariantTest, opaqueSerializationNotRegistered) {
       std::make_shared<opaqueSerializationTestStruct>());
   EXPECT_THROW(opaqueAfterRegistration.serialize(), VeloxException);
 }
+
+TEST(VariantTest, toJson) {
```

Comment:
Add two negative test cases.
```
auto rowTypePtr = ROW({{"c0", DECIMAL(10, 3)}});
EXPECT_THROW(
      variant::row({variant((int128_t)123456789)}).toJson(rowTypePtr), "Not same TypeKind");
rowTypePtr = ROW({{"c0", DECIMAL(20, 3)}});
EXPECT_THROW(
      variant::row({variant((int64_t)123456789)}).toJson(rowTypePtr), "Not same TypeKind");
```

Reply:
Added the provided negative tests.
```
  EXPECT_THROW(
      variant::row({variant((int128_t)123456789)}).toJson(rowTypePtr),
      VeloxUserError);
  rowTypePtr = ROW({{"c0", DECIMAL(20, 3)}});
  EXPECT_THROW(
      variant::row({variant((int64_t)123456789)}).toJson(rowTypePtr),
      VeloxUserError);
```

Reply:
Tests ran fine with the new negative cases as expected.
```
Running main() from /root/Presto/velox/_build/debug/_deps/gtest-src/googletest/src/gtest_main.cc
Note: Google Test filter = VariantTest.toJson
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from VariantTest
[ RUN      ] VariantTest.toJson
WARNING: Logging before InitGoogleLogging() is written to STDERR
E0626 08:52:28.771492  8439 Exceptions.h:68] Line: /root/Presto/velox/velox/type/Variant.cpp:229, Function:toJson, Expression: v.kind() == type->childAt(idx)->kind() (HUGEINT vs. BIGINT) Not same TypeKind, Source: USER, ErrorCode: INVALID_ARGUMENT
E0626 08:52:28.780982  8439 Exceptions.h:68] Line: /root/Presto/velox/velox/type/Variant.cpp:229, Function:toJson, Expression: v.kind() == type->childAt(idx)->kind() (BIGINT vs. HUGEINT) Not same TypeKind, Source: USER, ErrorCode: INVALID_ARGUMENT
[       OK ] VariantTest.toJson (9 ms)
[----------] 1 test from VariantTest (9 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test suite ran. (10 ms total)
[  PASSED  ] 1 test.
```